### PR TITLE
Terraform reference docs

### PR DESCRIPTION
**Pull request by gzigzigzeo**
Closes teleport-plugins/#253

### CODE REVIEW

File: docs/pages/setup/guides/terraform-provider/reference.mdx [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+```
+provider "teleport" {
+  addr         = "localhost:3025"
+  cert_path    = "tf.crt"
+  key_path     = "tf.key"
+  root_ca_path = "tf.ca"
+}
+```
+
+# <a name="teleport_user"></a>`teleport_user`
```
Comment (inertial-frame) [inline docs/pages/setup/guides/terraform-provider/reference.mdx:62]: Great job splitting up each of these sections! Super easy to read through and find what you need.

Nitpicks:

1. Can you remove the anchor tags? Links to the H1, H2, ... headings will be automatically generated.
2. Can you remove tic marks from the headings?

File: docs/pages/setup/guides/terraform-provider/reference.mdx [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+
+User resource allows to manage Teleport users: create, update, delete, assign and revoke roles and permissions.
```
Comment (inertial-frame) [inline docs/pages/setup/guides/terraform-provider/reference.mdx:64]: Nit: `create`, `update`, `delete`, `assign`, and `revoke` - reduces ambiguity so it's clear that `assign and revoke` is not a role. Thanks!

File: docs/pages/setup/guides/terraform-provider/reference.mdx [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+
+## `metadata`
+
+`metadata` is a user metadata
```
Comment (inertial-frame) [inline docs/pages/setup/guides/terraform-provider/reference.mdx:68]: Nit: colon or period here please

File: docs/pages/setup/guides/terraform-provider/reference.mdx [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+
+    saml_identities {
+      connector_id = "saml.example.com"
+      username     = "example"
+    }
+  }
+}
+```
+
+# <a name="teleport_role"></a>`teleport_role`
```
Comment (inertial-frame) [inline docs/pages/setup/guides/terraform-provider/reference.mdx:153]: Same as above thanks

File: docs/pages/setup/guides/terraform-provider/reference.mdx [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+
+Metadata is a role metadata
```
Comment (inertial-frame) [inline docs/pages/setup/guides/terraform-provider/reference.mdx:155]: Colon or period here and in similar places below please

File: docs/pages/setup/guides/terraform-provider/reference.mdx [hunk 6]
```diff
@@ -0,0 +1,422 @@
...
+|    `claims_to_roles.value`|      string | Claim value to match                                            |
+|               `thresholds`|         set | List of thresholds, one of which must be met in order for reviews to trigger a state-transition. If no thresholds are provided, a default threshold of 1 for approval and denial is used. |
+|          `thresholds.name`|      string | Threshold name                                                  |
+|       `thresholds.approve`|      number | Number of matching approvals needed for state-transition        |
+|          `thresholds.deny`|      number | Number of denials needed for state-transition                   |
+|        `thresholds.filter`|      string | Optional predicate used to determine which reviews count toward this threshold. |
+
+#### `spec.allow.review_requests` and `spec.deny.review_requests`
+
+`review_requests` is a matcher for allow/deny restrictions on  access reviews.
```
Comment (inertial-frame) [inline docs/pages/setup/guides/terraform-provider/reference.mdx:227]: Extra space. Also, `allow/deny` implies that these are interchangeable or the same. Can we say `allow`, `deny` instead?

File: docs/pages/setup/guides/terraform-provider/reference.mdx [hunk 7]
```diff
@@ -0,0 +1,422 @@
...
+|      `oidc_identities`| object list | List of OpenID Connect identities                                 |
+| `oidc_identities.connector_id` | string | An id of registered OIDC connector, e.g. 'google-example.com' |
+| `oidc_identities.username` | string | OIDC username                                                     |
+|      `saml_identities`| object list | List of SAML identities                                           |
+| `saml_identities.connector_id` | string | An id of registered SAML connector, e.g. 'google-example.com' |
+| `saml_identities.username` | string | SAML username                                                     |
+
+Example:
+
+```
```
Comment (inertial-frame) [inline docs/pages/setup/guides/terraform-provider/reference.mdx:115]: Do you mind adding the language: `YAML` - I'm not sure this presently has any functional differences but we've planned some cool things in the future that might use these more

File: docs/pages/setup/guides/terraform-provider/reference.mdx [hunk 8]
```diff
@@ -0,0 +1,422 @@
...
+|-------------------------|------------|----------|-------------------------------------------|
+|                   `name`|     string |        * | Token name                                |
+|            `description`|     string |          | Token description                         |
+|                 `labels`| string map |          | Resource labels                           |
+|              `namespace`|     string |          | Resource namespace ("default" by default) |
+|                `expires`|     string |        * | Expiry time in RFC8829 format             |
+
+## `spec`
+
+`spec` is a provision token spec. It has a single field: `roles` (string list) which is a list of roles associated with the token,  that will be converted to metadata in the SSH and X509 certificates issued to the user of the token.
```
Comment (inertial-frame) [inline docs/pages/setup/guides/terraform-provider/reference.mdx:275]: Probably break this up into two sentences. `. It will be ...`

**Review by inertial-frame — CHANGES_REQUESTED** [review]
Thank you very much! Super helpful! 

Just a few editorial items here - I will validate further once changes are made.

**Comment by klizhentas** [comment]
I will send a different PR with this as a base.

**Pull request closed without merging**